library/core/src/panicking.rs

```diff
@@ -205,8 +208,8 @@ fn panic_bounds_check(index: usize, len: usize) -> ! {
     panic!("index out of bounds: the len is {len} but the index is {index}")
 }
 
-#[cold]
-#[cfg_attr(not(feature = "panic_immediate_abort"), inline(never))]
+#[cfg_attr(not(feature = "panic_immediate_abort"), inline(never), cold)]
+#[cfg_attr(feature = "panic_immediate_abort", inline)]
```

Comment:
@saethlin is there any reason you used different attributes for this function than what all the others in this file do?

Reply:
https://github.com/rust-lang/rust/issues/118393